.github/workflows/ci.yml

```diff
@@ -0,0 +1,445 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - staging
+      - feature*
+
+env:
+  # Caching
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_DEFAULT_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  CACHIX_SIGNING_KEY: ${{ secrets.CACHIX_SIGNING_KEY }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NIX_CACHE_PRIV_KEY: ${{ secrets.NIX_CACHE_PRIV_KEY }}
+  NIX_CACHE_PUB_KEY: ${{ secrets.NIX_CACHE_PUB_KEY }}
+  NIX_CONFIG: access-tokens = github.com=${{ secrets.NIXPKGS_PRIVATE_PAT }}
+
+  # Other
+  GH_PROJECT_PATH: ${{ github.repository }}
+  GIT_SUBMODULE_STRATEGY: recursive
+  npm_config_cache: "${{ github.workspace }}/tmp/npm"
+  npm_config_prefer_offline: "true"
+
+jobs:
+  check-lint:
+    name: "Check / Lint"
```

Comment:
It is consistent, I just decided to use `-` instead of `:` because its more in line with how YAML keys are named. The other part `name:`, that is just the display name on GitHub, it does not change anything, but looks more readable. Example:
![image](https://github.com/MatrixAI/Polykey-CLI/assets/49110391/f626b859-b51c-4c68-9bb2-d7328acf1e0a)